holoviews/core/util.py

```diff
@@ -24,6 +26,57 @@
 except ImportError:
     dd = None
 
+
+
+
+class HashGenerator(json.JSONEncoder):
+    """
+    Extends JSONEncoder to generate a hashable string for as many types
+    of key as possible. These keys are supplied by widgets and Stream
+    parameters and may be nested. If an unrecognized object type is
```

Comment:
> I don't think the docstring for this class should mention widgets or Streams in any way other than something explicitly labeled as an example. It could mention memoization, though, so that we know the goal of this object is to generate an id that can tell whether two objects differ, not to provide an encoding of the object.

Happy to update the docstring as suggested.